main.js

```diff
@@ -0,0 +1,79 @@
+const apiKey = "ee02afd279ba852f5518dbe94195f595";
+
+const searchButton = document.querySelector(".btn-primary");
+const cityInput = document.querySelector('.form-control');
+const weatherContainer = document.getElementById('weather-container');
+const currentWeatherDiv = document.querySelector('.current-weather');
+const forecastDiv = document.querySelector('.forecast');
+
+currentWeatherDiv.style.display = 'none';
+forecastDiv.style.display = 'none';
+
+searchButton.addEventListener('click', () => {
+  const city = cityInput.value;
+  if (!city) {
+   alert('No city name given');
+   return;
+  }
+  fetchCurrentWeather(city);
+  fetchForecast(city);
+});
+
+function fetchCurrentWeather(city) {
+  const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=imperial&appid=${apiKey}`;
+  fetch(url)
+    .then((response) => response.json())
+    .then(function (data) {
+      const cityNameElement = document.getElementById('city-name');
+      const weatherIconElement = document.getElementById('current-weather-icon');
+      const tempElement = document.getElementById('current-temp');
+
+      cityNameElement.textContent = data.name;
+      weatherIconElement.src = `https://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png`;
+      weatherIconElement.alt = data.weather[0].description;
+      tempElement.textContent = `${data.main.temp} °F - ${data.weather[0].description}`;
+
+      currentWeatherDiv.style.display = 'block';
+    })
+    .catch((error) => {
+      console.error('Error, didn\'t fetch weather', error);
+    });
+}
+
+function fetchForecast(city) {
```

Comment:
this function is too big. Need to be extracted into smaller chunks to help with readability and maintainability

Reply:
Thanks, Rony. How would this look as an example?

Reply:
the function is getting data and adding it to the dom, you can create a get data function that returns a promise of that data and then split the for loop logic as well. 
The best you can do is ask yourself, is this function only fetching the forecast? not only, is fetching and... rendering, and arranging the data and displaying, etc... all these can be functions. I suggest you review this deeper with your mentor